packages/stryker/src/initializer/StrykerInitializer.ts

```diff
@@ -0,0 +1,133 @@
+import * as child from 'child_process';
+import { StrykerInquirer } from './StrykerInquirer';
+import NpmClient from './NpmClient';
+import PromptOption from './PromptOption';
+import * as log4js from 'log4js';
+import { filterEmpty } from '../utils/objectUtils';
+import StrykerConfigWriter from './StrykerConfigWriter';
+
+const log = log4js.getLogger('StrykerInitializer');
+
+export default class StrykerInitializer {
+
+  private inquirer = new StrykerInquirer();
+
+  constructor(private out = console.log, private client: NpmClient = new NpmClient()) { }
+
+  /**
+   * Runs the initializer ask used framework and test runner en setup environment
+   * @function
+   */
+  async initialize(): Promise<void> {
+    StrykerConfigWriter.guardForExistingConfig();
+    this.patchProxies();
+    const selectedTestRunner = await this.selectTestRunner();
+    const selectedTestFramework = selectedTestRunner ? await this.selectTestFramework(selectedTestRunner) : null;
+    const selectedReporters = await this.selectReporters();
+    const npmDependencies = this.getSelectedNpmDependencies(selectedTestRunner, selectedTestFramework, selectedReporters);
+    this.installNpmDependencies(npmDependencies);
+    await new StrykerConfigWriter(this.out,
+      selectedTestRunner,
```

Comment:
Let's leave it like it is for now. This class is only used once. I've added it because of a remark from @korthout regarding clean code.